packages/core-data/src/resolvers.js

```diff
@@ -707,3 +707,168 @@ export const getNavigationFallbackId =
 			] );
 		}
 	};
+
+/**
+ * Requests an entity's revisions from the REST API.
+ *
+ * @param {string}           kind     Entity kind.
+ * @param {string}           name     Entity name.
+ * @param {number|string}    parentId Record's key whose revisions you wish to fetch.
+ * @param {Object|undefined} query    Optional object of query parameters to
+ *                                    include with request. If requesting specific
+ *                                    fields, fields must always include the ID.
+ */
+export const getRevisions =
+	( kind, name, parentId, query = {} ) =>
+	async ( { dispatch } ) => {
+		const configs = await dispatch( getOrLoadEntitiesConfig( kind ) );
+		const entityConfig = configs.find(
+			( config ) => config.name === name && config.kind === kind
+		);
+
+		if (
+			! entityConfig ||
+			entityConfig?.__experimentalNoFetch ||
+			! entityConfig?.supports?.revisions
+		) {
+			return;
+		}
+
+		const lock = await dispatch.__unstableAcquireStoreLock(
+			STORE_NAME,
+			[ 'entities', 'records', kind, name, 'revisions', parentId ],
+			{ exclusive: false }
+		);
+
+		try {
+			if ( query._fields ) {
+				// If requesting specific fields, items and query association to said
+				// records are stored by ID reference. Thus, fields must always include
+				// the ID.
+				query = {
+					...query,
+					_fields: [
+						...new Set( [
+							...( getNormalizedCommaSeparable( query._fields ) ||
+								[] ),
+							entityConfig.key || DEFAULT_ENTITY_KEY,
```

Comment:
> Is the revisionKey always the same as the parent recordKey? 

Yeah, as far as revisions are concerned the key property will always be `id`, so for the parent `{ id: revisionKey }` and the revision `{ id: recordKey, parent: revisionKey }`

I copied this over 1:1 from `getEntityRecord`, so it looks like we don't really need it. 

> And are there even any entities that would have a non-default (other than id) key and support revisions? 

Looking at the [rootEntitiesConfig](https://github.com/WordPress/gutenberg/blob/try%2Fcore-data-entity-revisions/packages/core-data/src/entities.js#L27-L27), I can't see any that would support revisions.

However, the [globalStyles root kind](https://github.com/WordPress/gutenberg/blob/try%2Fcore-data-entity-revisions/packages/core-data/src/entities.js#L206-L206) doesn't have a `key: 'id;` entry so I expect it would use the default `DEFAULT_ENTITY_KEY`.

 I think we could probably add the `key` to the globalStyles entity though.

Reply:
I've updated it to just use `DEFAULT_ENTITY_KEY` 👍🏻